src/ralph/backends/data/base.py

```diff
@@ -553,25 +555,34 @@ async def close(self) -> None:
 
 
 def get_backend_generic_argument(
-    backend_class: Type[Union[BaseDataBackend, BaseAsyncDataBackend]], position: int
+    backend_class: Type[Union[BaseDataBackend, BaseAsyncDataBackend]],
+    position: DataBackendArgument,
 ) -> Optional[Type]:
     """Return the generic argument of `backend_class` at specified `position`."""
-    if not hasattr(backend_class, "__orig_bases__"):
-        return None
+    for base in get_original_bases(backend_class):
+        if not (
+            hasattr(base, "__origin__")
+            and isclass(base.__origin__)
+            and issubclass(base.__origin__, (BaseDataBackend, BaseAsyncDataBackend))
```

Comment:
Here, could we use [`get_origin`](https://docs.python.org/3.12/library/typing.html#typing.get_origin)?

Reply:
Well spotted) Thanks) Updated it using get_origin)